src/environment.js

```diff
@@ -324,7 +325,9 @@ var Environment = Obj.extend({
 
         var tmpl = new Template(src, this, opts.path);
         return tmpl.render(ctx, cb);
-    }
+    },
+
+    waterfall: waterfall
```

Comment:
I don't have an issue with this if it works :-) What downside do you see that prompted this comment?